### ISSUE TITLE
file.serialize - merge_if_exists with json - crash

### ISSUE THREAD

**Issue by cr1st1p** [saltstack/salt#30572]
OS: Ubuntu 15.10
Salt - installed from standard repository:  2015.5.3

In short: state.file.serialize uses wrong json.load instead of json.loads 

I have a state like:
name:
```sls
  file.serialize:
    - merge_if_exists: true
    - formatter: json
```
When run, I get:
>     Comment: An exception occurred in this state: Traceback (most recent call last):
>               File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1560, in call                                                              
>                  **cdata['kwargs'])                                                                                                                   
>                 File "/usr/lib/python2.7/dist-packages/salt/states/file.py", line 4015, in serialize                                                   
>                  existing_data = json.load(fhr.read())                                                                                                
>                File "/usr/lib/python2.7/json/__init__.py", line 286, in load                                                                          
>                  return loads(fp.read(),                                                                                                              
>              AttributeError: 'str' object has no attribute 'read'                                                                                     
>     Started: 08:48:08.941196
>    Duration: 2.501 ms
>     Changes:   
>

Cause is the wrong call to json.load(fhr.read() instead of json.loads(fhr.read
I manually changed it locally and then it worked ok.

**Comment by TronPaul** [comment]
Probably the better solution is to call `json.load(fhr)` rather than `json.loads(fhr.read())`.

**Comment by jfindlay** [comment]
@cr1st1p, thanks for the report.  This has been fixed in d12be523.

**Comment by rallytime** [comment]
Closing since this issue is fixed. If you have any more questions or this issue comes up again, leave a comment and we will be happy to re-open!
